src/main/java/com/parsing/parsers/pdf/parsing/DataRecognizer.java

```diff
@@ -0,0 +1,156 @@
+    private BigDecimal totalPrice = null;
+
+    public final boolean recognizeLotPDFResult(List<Row> rows) {
+        boolean result = false;
+        for (Row row : rows) {
+            if (isModelRow(row)) {
+                int modelColumnNumber = getModelColumnNumber(row);
+                List<String> amounts = new ArrayList<>();
+                List<String> prices = new ArrayList<>();
+                for (int x = modelColumnNumber; x < row.getColumns().size(); x++) {
+                    if (findAmount(row.getColumns().get(x).getParsingResult()) != null) {
+                        amounts.add(findAmount(row.getColumns().get(x).getParsingResult()));
+                    }
+                    if (findPrices(row.getColumns().get(x).getParsingResult()) != null) {
+                        prices.add(findPrices(row.getColumns().get(x).getParsingResult()));
+                    }
+                }
+                getAmount(amounts);
+                getPriceAndTotalPrice(prices);
+                result = saveItems();
+            }
+        }
+        return result;
+    }
+
+    private boolean saveItems() {
+        if (totalPrice != null && totalPrice.equals(BigDecimal.valueOf(amount).multiply(price))) {
+            lotPDFResultService.saveLaptopItem(model, price, amount);
+            return true;
+        }
+        return false;
+    }
+
+    private void getAmount(List<String> amounts) {
+        if (amounts.size() == 1) {
+            amount = Integer.parseInt(amounts.get(0));
+        } else if (amounts.size() > 1) {
+            log.debug("more than one amount number found in one row");
+            amount = 0;
+        }
+        if (amount == 0 && price != null) {
+            amount = totalPrice.divide(price).toBigInteger().intValueExact();
+        }
+    }
+
+    private void getPriceAndTotalPrice(List<String> prices) {
+        List<BigDecimal> bigDecimals = null;
+        try {
+            bigDecimals = prices.stream().map(BigDecimal::new).toList();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        Optional<BigDecimal> min = bigDecimals.stream().min(Comparator.naturalOrder());
+        Optional<BigDecimal> max = bigDecimals.stream().max(Comparator.naturalOrder());
+        if (min.isPresent() && max.isPresent()) {
+            price = min.get();
+            totalPrice = max.get();
+        }
+    }
+
+    private int getModelColumnNumber(Row row) {
+        int modelColumnNumber = 0;
+        for (Column column : row.getColumns()) {
+            if (isModel(column.getParsingResult())) {
+                model = findModel(column.getParsingResult());
+                modelColumnNumber = row.getColumns().indexOf(column);
+            }
+        }
+        return modelColumnNumber;
+    }
+
+    private boolean isModelRow(Row row) {
+        List<String> listLowerCaseColumn = row.getColumns().stream().map(x -> x.getParsingResult().toLowerCase(Locale.ROOT)).toList();
```

Comment:
looks to long for me, consider making the line shorter.

Reply:
Done.